homeassistant/components/transmission/__init__.py

```diff
@@ -206,6 +224,34 @@ def add_torrent(service):
                     "Could not add torrent: unsupported type or no permission"
                 )
 
+        def start_torrent(service):
+            """Start torrent."""
+            tm_client = None
+            for entry in self.hass.config_entries.async_entries(DOMAIN):
+                if entry.data[CONF_NAME] == service.data[CONF_NAME]:
+                    tm_client = self.hass.data[DOMAIN][entry.entry_id]
+                    break
+            if tm_client is None:
+                _LOGGER.error("Transmission instance is not found")
+                return
+            torrent_id = service.data[CONF_ID]
+            tm_client.tm_api.start_torrent(torrent_id)
+            tm_client.api.update()
+
+        def stop_torrent(service):
+            """Stop torrent."""
+            tm_client = None
+            for entry in self.hass.config_entries.async_entries(DOMAIN):
```

Comment:
Almost everything about the two service handlers are the same for both services. Only one api call differs. Please refactor and make a helper or similar to consolidate this code.